src/CoreWCF.NetTcp/tests/AuthorizationTest.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ServiceModel.Channels;
+using CoreWCF.Configuration;
+using CoreWCF.Description;
+using CoreWCF.IdentityModel.Policy;
+using CoreWCF.Primitives.Tests.CustomSecurity;
+using CoreWCF.Security;
+using Helpers;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CoreWCF.NetTcp.Tests
+{
+    public class AuthorizationTest
+    {
+        private readonly ITestOutputHelper _output;
+
+        public AuthorizationTest(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [Fact(Skip ="Skipped in pipeline run")]
+        public void AuthorizationBasedonRolesTest()
+        {
+            string testString = "a" + PrincipalPermissionMode.Always.ToString() + "test";
+            IWebHost host = ServiceHelper.CreateWebHostBuilder<StartupForAuthorization>(_output).Build();
+            using (host)
+            {
+                System.ServiceModel.ChannelFactory<ClientContract.ITestService> factory = null;
+                ClientContract.ITestService channel = null;
+                host.Start();
+                try
+                {
+                    System.ServiceModel.NetTcpBinding binding = ClientHelper.GetBufferedModeBinding(System.ServiceModel.SecurityMode.Transport);
+                    factory = new System.ServiceModel.ChannelFactory<ClientContract.ITestService>(binding,
+                        new System.ServiceModel.EndpointAddress(host.GetNetTcpAddressInUse() + Startup.WindowsAuthRelativePath));
+                    channel = factory.CreateChannel();
+                    ((IChannel)channel).Open();
+                    string result = channel.EchoForAuthorizarionOneRole(testString);
+                    Assert.Equal(testString, result);
+                    try
+                    {
+                        channel.EchoForAuthorizarionNoRole(testString);
+                    }catch(Exception ex)
+                    {
+                        Assert.Contains("Access is denied", ex.Message);
+                    }
```

Comment:
Do you always expect this exception to be thrown? If the method doesn't throw an exception, then the Assert doesn't get hit and the test fails. Should this be Assert.ThrowsAny?

Reply:
This is expected to be thrown. I wanted to validate the message.